src/Datadog.Trace.ClrProfiler.Native/pal.h

```diff
@@ -37,34 +40,29 @@ inline WSTRING DatadogLogFilePath(const std::string& file_name_suffix)
 #else
                WStr('/') +
 #endif
-               ToWSTRING("dotnet-tracer-native" + file_name_suffix + ".log");
+               ToWSTRING(file_name);
     }
 
-    WSTRING path = GetEnvironmentValue(environment::log_path);
+    WSTRING path = GetEnvironmentValue(TLoggerPolicy::logging_environment::log_path);
 
     if (path.length() > 0)
     {
         return path;
     }
 
 #ifdef _WIN32
-    char* p_program_data;
-    size_t length;
-    const errno_t result = _dupenv_s(&p_program_data, &length, "PROGRAMDATA");
-    std::string program_data;
-
-    if (SUCCEEDED(result) && p_program_data != nullptr && length > 0)
-    {
-        program_data = std::string(p_program_data);
-    }
-    else
+    WSTRING program_data = GetEnvironmentValue(WStr("PROGRAMDATA"));
+    std::filesystem::path program_data_path;
+    if (program_data.length() == 0)
     {
-        program_data = R"(C:\ProgramData)";
+        program_data_path = WStr(R"(C:\ProgramData)");
     }
 
-    return ToWSTRING(program_data + R"(\Datadog .NET Tracer\logs\dotnet-tracer-native)" + file_name_suffix + ".log");
+    // on Windows WSTRING == wstring
+    return (program_data_path / TLoggerPolicy::product_folder_name / WStr("logs") / file_name).wstring();
 #else
-    return ToWSTRING("/var/log/datadog/dotnet/dotnet-tracer-native" + file_name_suffix + ".log");
+    // TODO: should we consider having a logs folder on non-Windows ?
```

Comment:
```
// TODO: should we consider having a logs folder on non-Windows ?
```
In Windows we create a `logs` folder because there could potentially be other files in the parent folder (`C:\ProgramData\Datadog .NET Tracer`), like config files or other data. See `C:\ProgramData\Datadog` and `C:\ProgramData\Datadog\logs`, for example.

On Linux, this folder is in `/var/log`, so it's expected to only contain log files, right?

Reply:
🤔 good one. `/var/log` is expected to contain log files